File: frontend/types/validation-evidence.ts
import type { ProjectDocument } from "@/types/document";

export type ValidationEvidenceCategory = "technical" | "economics" | "logistics" | "power_h2" | "commercial_mrv";
export type ValidationEvidenceStatus = "missing" | "requested" | "received" | "verified" | "rejected";
export type ValidationEvidencePriority = "high" | "medium" | "low";
export type ValidationEvidenceConfidence = "high" | "medium" | "low" | "unknown";

export type ValidationEvidenceRecord = {
  id: string;
  plant_id: string;
  scenario_id: string;
  document_id: string | null;
  category: ValidationEvidenceCategory;
  evidence_key: string;
  title: string;
  required_evidence: string | null;
  current_basis: string | null;
  status: ValidationEvidenceStatus;
  priority: ValidationEvidencePriority;
  owner_name: string | null;
  source_organization: string | null;
  reference_url: string | null;
  due_date: string | null;
  received_date: string | null;
  verified_date: string | null;
  confidence_level: ValidationEvidenceConfidence;
  notes: string | null;
  supporting_document: ProjectDocument | null;
  created_at: string;
  updated_at: string;
};

export type ValidationEvidencePayload = {
  plant_id?: string;
  scenario_id?: string;
  document_id?: string | null;
  category?: ValidationEvidenceCategory;
  evidence_key?: string;
  title?: string;
  required_evidence?: string | null;
  current_basis?: string | null;
  status?: ValidationEvidenceStatus;
  priority?: ValidationEvidencePriority;
  owner_name?: string | null;
  source_organization?: string | null;
  reference_url?: string | null;
  due_date?: string | null;
  received_date?: string | null;
  verified_date?: string | null;
  confidence_level?: ValidationEvidenceConfidence;
  notes?: string | null;
};

export type ValidationEvidenceRequirement = {
  plant_id: string;
  scenario_id: string;
  category: ValidationEvidenceCategory;
  evidence_key: string;
  title: string;
  required_evidence: string;
  current_basis: string;
  priority: ValidationEvidencePriority;
  source_validation_status: string;
  status: ValidationEvidenceStatus;
  confidence_level: ValidationEvidenceConfidence;
  status_score: number;
  record: ValidationEvidenceRecord | null;
};

export type ValidationEvidenceCandidate = {
  validation_rank: number;
  site_name: string;
  plant_id: string;
  scenario_id: string;
  province: string | null;
  capacity_mw: number | null;
  final_score: number;
  methanol_tpy: number | null;
  estimated_lcom_usd_ton: number | null;
  nearest_port_name: string | null;
  evidence_readiness_score: number;
  high_priority_open_items: number;
  evidence_items: ValidationEvidenceRequirement[];
};

export type ValidationEvidenceWorkspace = {
  scheme: string;
  limit: number;
  summary: {
    candidate_count: number;
    lead_candidate: string | null;
    total_items: number;
    verified_items: number;
    high_priority_open_items: number;
    evidence_readiness_score: number;
    counts_by_status: Record<ValidationEvidenceStatus, number>;
  };
  candidates: ValidationEvidenceCandidate[];
  warnings: string[];
};

File: backend/app/schemas/validation_evidence.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import ConfidenceLevel
from app.schemas.document import DocumentRead


ValidationEvidenceCategory = Literal["technical", "economics", "logistics", "power_h2", "commercial_mrv"]
ValidationEvidenceStatus = Literal["missing", "requested", "received", "verified", "rejected"]
ValidationEvidencePriority = Literal["high", "medium", "low"]

VALIDATION_EVIDENCE_CATEGORY_VALUES = ("technical", "economics", "logistics", "power_h2", "commercial_mrv")
VALIDATION_EVIDENCE_STATUS_VALUES = ("missing", "requested", "received", "verified", "rejected")
VALIDATION_EVIDENCE_PRIORITY_VALUES = ("high", "medium", "low")


class ValidationEvidenceCreate(BaseModel):
    plant_id: str
    scenario_id: str
    document_id: str | None = None
    category: ValidationEvidenceCategory
    evidence_key: str = Field(min_length=1, max_length=120)
    title: str = Field(min_length=1, max_length=220)
    required_evidence: str | None = None
    current_basis: str | None = None
    status: ValidationEvidenceStatus = "missing"
    priority: ValidationEvidencePriority = "high"
    owner_name: str | None = Field(default=None, max_length=160)
    source_organization: str | None = Field(default=None, max_length=160)
    reference_url: str | None = Field(default=None, max_length=500)
    due_date: date | None = None
    received_date: date | None = None
    verified_date: date | None = None
    confidence_level: ConfidenceLevel = "unknown"
    notes: str | None = None


class ValidationEvidenceUpdate(BaseModel):
    document_id: str | None = None
    title: str | None = Field(default=None, min_length=1, max_length=220)
    required_evidence: str | None = None
    current_basis: str | None = None
    status: ValidationEvidenceStatus | None = None
    priority: ValidationEvidencePriority | None = None
    owner_name: str | None = Field(default=None, max_length=160)
    source_organization: str | None = Field(default=None, max_length=160)
    reference_url: str | None = Field(default=None, max_length=500)
    due_date: date | None = None
    received_date: date | None = None
    verified_date: date | None = None
    confidence_level: ConfidenceLevel | None = None
    notes: str | None = None


class ValidationEvidenceRead(BaseModel):
    id: str
    plant_id: str
    scenario_id: str
    document_id: str | None
    category: ValidationEvidenceCategory
    evidence_key: str
    title: str
    required_evidence: str | None
    current_basis: str | None
    status: ValidationEvidenceStatus
    priority: ValidationEvidencePriority
    owner_name: str | None
    source_organization: str | None
    reference_url: str | None
    due_date: date | None
    received_date: date | None
    verified_date: date | None
    confidence_level: ConfidenceLevel
    notes: str | None
    supporting_document: DocumentRead | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ValidationEvidenceRequirement(BaseModel):
    plant_id: str
    scenario_id: str
    category: ValidationEvidenceCategory
    evidence_key: str
    title: str
    required_evidence: str
    current_basis: str
    priority: ValidationEvidencePriority
    source_validation_status: str
    status: ValidationEvidenceStatus
    confidence_level: ConfidenceLevel
    status_score: float
    record: ValidationEvidenceRead | None = None


class ValidationEvidenceCandidate(BaseModel):
    validation_rank: int
    site_name: str
    plant_id: str
    scenario_id: str
    province: str | None
    capacity_mw: float | None
    final_score: float
    methanol_tpy: float | None
    estimated_lcom_usd_ton: float | None
    nearest_port_name: str | None
    evidence_readiness_score: float
    high_priority_open_items: int
    evidence_items: list[ValidationEvidenceRequirement]


class ValidationEvidenceSummary(BaseModel):
    candidate_count: int
    lead_candidate: str | None
    total_items: int
    verified_items: int
    high_priority_open_items: int
    evidence_readiness_score: float
    counts_by_status: dict[str, int]


class ValidationEvidenceWorkspace(BaseModel):
    scheme: str
    limit: int
    summary: ValidationEvidenceSummary
    candidates: list[ValidationEvidenceCandidate]
    warnings: list[str]
